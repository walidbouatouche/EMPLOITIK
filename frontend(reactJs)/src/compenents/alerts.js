import React, { useState } from 'react'

export const Alerts = {
    AlertDanger,
    Alertsuccess
}

function AlertDanger({ text }) {
    const [display, setDisplay] = useState("block")
    setTimeout(() => {
        setDisplay("none")
    }, 1000)
    return (
        <div className="w3-panel w3-pale-red w3-padding-16 w3-animate-top" style={{ display: display }}>
            <p>{text}</p>
        </div>)
}
function Alertsuccess({ text }) {
    const [display, setDisplay] = useState("block")
    setTimeout(() => {
        setDisplay("none")
    }, 1000)
    return (
        <div className="w3-panel w3-pale-green w3-padding-16  w3-animate-top" style={{ display: display }}>
            <p>{text}</p>
        </div>)
}



