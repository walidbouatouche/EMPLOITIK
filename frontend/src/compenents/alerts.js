import React from 'react'

export const Alerts = {
    AlertDanger,
    Alertsuccess
}
function AlertDanger({ text }) {
    return (
        <div className="w3-panel w3-pale-red w3-padding">
            <p>{text}</p>
        </div>)
}
function Alertsuccess({ text }) {
    return (
        <div className="w3-panel w3-pale-green w3-padding">
            <p>{text}</p>
        </div>)
}



