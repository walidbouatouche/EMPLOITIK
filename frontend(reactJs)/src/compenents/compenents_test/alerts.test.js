import React from 'react'
import { Alerts } from '../alerts'
import toJson from 'enzyme-to-json';
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { render, fireEvent, cleanup } from '@testing-library/react'
Enzyme.configure({ adapter: new Adapter() })
const wrapperDanger = shallow(<Alerts.AlertDanger />)
const wrapperSuccess = shallow(<Alerts.Alertsuccess />)

it(' Alerts render  correctly ', () => {

    expect(toJson(wrapperDanger)).toMatchSnapshot();
    expect(toJson(wrapperSuccess)).toMatchSnapshot();

});
 


