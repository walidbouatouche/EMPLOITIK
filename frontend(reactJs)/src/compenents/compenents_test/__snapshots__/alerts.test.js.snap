// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Alerts render  correctly  1`] = `
<div
  className="w3-panel w3-pale-red w3-padding-16 w3-animate-top"
  style={
    Object {
      "display": "block",
    }
  }
>
  <p />
</div>
`;

exports[` Alerts render  correctly  2`] = `
<div
  className="w3-panel w3-pale-green w3-padding-16  w3-animate-top"
  style={
    Object {
      "display": "block",
    }
  }
>
  <p />
</div>
`;
